frontend/src/lib/components/Badge.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { CheckOutlined, WarningOutlined } from '@ant-design/icons'
+import { Tooltip } from 'antd'
+
+interface BadgeProps {
+    icon?: JSX.Element
+    type?: 'success' | 'warning' | 'danger'
+    className?: string
+    onClick?: () => void
+    tooltip?: string
+}
+
+export function Badge({ icon, type, className, onClick, tooltip }: BadgeProps): JSX.Element {
+    const getTypeIcon = (): JSX.Element | undefined => {
+        if (type === 'success') {
+            return <CheckOutlined />
+        } else if (type === 'warning') {
+            return <WarningOutlined />
+        } else if (type === 'danger') {
+            return <WarningOutlined />
+        }
```

Comment:
Nit: could become `return type === 'success' ? <CheckOutlined /> : <WarningOutlined />`

Or is there a reason this doesn't have a default return?

Reply:
Well the only reason for this would be to future-proof for more types of badges

Reply:
I mean, wouldn't be too complicated to add that logic then - but it's far from important to me to use the ternary operator here. 

The main question is about why not have a default return making it so that it may return `undefined`

Reply:
Oh I see your point, well because we do have some badges that have no icon (i.e. expected behavior), I'll add a note there